### DISCUSSION TITLE
Bug with `packageRules` and `semanticCommitType`?

### DISCUSSION THREAD

**Discussion by Lamatte** [renovatebot/renovate#16357] [Request Help]
Hello.

I'm trying to handle a docker-compose file, and have Renovate generate:
- `fix` for minor docker images upgrades,
- `feat` for major ones. 

My current conf is :
```json
  "packageRules": [
    {   
      "matchDatasources": ["docker"],
      "major": {
        "semanticCommitType": "feat"
      },  
      "minor": {
        "semanticCommitType": "fix"
      }   
    }   
  ]
```

It seems to be working for all images but one (grafana).  
And it happens that this particular image has actually one minor AND one major update.  
Renovate properly reports a `feat` for the major, but reports a `chore` for the minor, where a `fix` was expected :
```
chore(deps): update dependency grafana/grafana to v8.5.6
fix(deps): update dependency prom/prometheus to v2.36.2
feat(deps): update dependency grafana/grafana to v9
```

Seems like a bug no ?

And by the way, Renovates warns me about:
```
Configuration Error: packageRules[0]: Each packageRule must contain at least one non-match* or non-exclude* field. Rule: {"matchDatasources":["docker"]}
```

Can't figure out what it means and how I can deal with that...

**Comment by rarkins** [comment]
Try with two rules and using matchUpdateTypes

**Reply by rarkins** [reply]
Ie don't nest major and minor objects within a package rule

**Reply by Lamatte** [reply]
Thanks @rarkins .

I tried:
```json
  "packageRules": [
    {   
      "matchDatasources": ["docker"],
      "matchUpdateTypes": ["major"],
      "extends": [":semanticCommitTypeAll(feat)"]
    },  
    {   
      "matchDatasources": ["docker"],
      "matchUpdateTypes": ["minor"],
      "extends": [":semanticCommitTypeAll(fix)"]
    }   
  ]
```

I no longer have the warning, but the minor upgrade of grafana still appears as a `chore` (and still OK for all the rest).

**Reply by Lamatte** [reply]
And FYI if I use
```
  "packageRules": [
    {   
      "matchDatasources": ["docker"],
      "extends": [":semanticCommitTypeAll(fix)"]
    }   
  ]
```
Renovate behaves as expected: both grafana upgrades appear as `fix`.

**Reply by rarkins** [reply]
Maybe the update type is "patch"?
